src/client.h

```diff
@@ -1,15 +1,34 @@
-#ifndef CLIENT_H
-#define CLIENT_H
+#pragma once
 
-#include "kodi/libXBMC_addon.h"
-#include "kodi/libXBMC_pvr.h"
-#include "p8-platform/threads/threads.h"
-#include "p8-platform/util/util.h"
-#include "windows.h"
+#include "Settings.h"
 
-extern std::string g_strUserPath;
-extern std::string g_strClientPath;
-extern ADDON::CHelper_libXBMC_addon *XBMC;
-extern CHelper_libXBMC_pvr *PVR;
-#endif
+#include <kodi/AddonBase.h>
+#include <unordered_map>
 
+class ZatData;
+
+class ATTRIBUTE_HIDDEN CZattooTVAddon : public kodi::addon::CAddonBase
+{
+public:
+  CZattooTVAddon() = default;
+
+  ADDON_STATUS Create() override;
+  ADDON_STATUS GetStatus() override { return m_curStatus; }
+  ADDON_STATUS SetSetting(const std::string& settingName,
+                          const kodi::CSettingValue& settingValue) override;
+  ADDON_STATUS CreateInstanceEx(int instanceType,
+                                const std::string& instanceID,
+                                KODI_HANDLE instance,
+                                KODI_HANDLE& addonInstance,
+                                const std::string& version) override;
+  void DestroyInstance(int instanceType,
+                       const std::string& instanceID,
+                       KODI_HANDLE addonInstance) override;
+
+  const CSettings& GetSettings() const { return m_settings; }
+
+private:
+  ADDON_STATUS m_curStatus = ADDON_STATUS_UNKNOWN;
+  CSettings m_settings;
+  std::unordered_map<std::string, ZatData*> m_usedInstances;
```

Comment:
As note about this, the instance becomes stored there, currently not direct needed. Thought only to allow the main addon class to communicate with running instances, if needed for something.

Also another feature which then can come in future, is that e.g. `ZatData` can be created several times for several backends. Thought after the rework from "C" to "C++" change is done to look about.